mypy/semanal.py

```diff
@@ -166,6 +166,23 @@
     'builtins.classmethod': 'classmethod.pyi',
 }  # type: Final
 
+# Map from the lowercased name of a missing definition to a tuple containing
+# the name of the likely module definition and a suggested import string.
+TYPES_FOR_UNIMPORTED_HINTS = {
+    'any': ('typing', 'from typing import Any'),
+    'callable': ('typing', 'from typing import Callable'),
+    'dict': ('typing', 'from typing import Dict'),
+    'iterable': ('typing', 'from typing import Iterable'),
+    'iterator': ('typing', 'from typing import Iterator'),
+    'list': ('typing', 'from typing import List'),
+    'optional': ('typing', 'from typing import Optional'),
+    'set': ('typing', 'from typing import Set'),
+    'tuple': ('typing', 'from typing import Tuple'),
+    'typevar': ('typing', 'from typing import TypeVar'),
+    'union': ('typing', 'from typing import Union'),
+    'cast': ('typing', 'from typing import cast'),
```

Comment:
Since all of these follow the same pattern, we could perhaps avoid some redundancy. One idea would be to just have a list/set of full names (`typing.Any`, `typing.Callable`, ...) and we can derive the other information (lower case version, `from typing import Foo`) from them.

Reply:
:+1: